tests/std/tests/P0896R4_P1614R2_spaceship/test.cpp

```diff
@@ -0,0 +1,193 @@
+    }
+    { // forward_list
+        std::forward_list<int> a1(3, 100);
+        std::forward_list<int> a2(3, 100);
+        std::forward_list<int> b1(2, 200);
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // map
+        std::map<std::string, int> a1;
+        a1["hi"]   = 1;
+        a1["hola"] = 2;
+        std::map<std::string, int> a2;
+        a2["hi"]   = 1;
+        a2["hola"] = 2;
+        std::map<std::string, int> b1;
+        b1["zoe"]   = 3;
+        b1["koala"] = 4;
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // multimap
+        std::multimap<char, int> a1 = {{'a', 1}, {'b', 2}, {'a', 3}};
+        std::multimap<char, int> a2 = {{'a', 1}, {'a', 3}, {'b', 2}};
+        std::multimap<char, int> b1 = {{'z', 4}, {'y', 90}, {'z', 12}};
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // set
+        std::set<int> a1;
+        a1.insert(10);
+        a1.insert(20);
+
+        std::set<int> a2;
+        a2.insert(10);
+        a2.insert(20);
+
+        std::set<int> b1;
+        b1.insert(30);
+        b1.insert(40);
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // multiset
+        std::multiset<int> a1;
+        a1.insert(10);
+        a1.insert(10);
+        a1.insert(20);
+
+        std::multiset<int> a2;
+        a2.insert(10);
+        a2.insert(20);
+        a2.insert(10);
+
+        std::multiset<int> b1;
+        b1.insert(30);
+        b1.insert(40);
+        b1.insert(40);
+        ordered_containers_test(a1, a2, b1);
+    }
+    { // unordered_map
+        using stringmap = std::unordered_map<std::string, std::string>;
+        stringmap a     = {{"cat", "tabby"}, {"dog", "poodle"}, {"bear", "grizzly"}};
+        stringmap b     = {{"dog", "poodle"}, {"bear", "grizzly"}, {"cat", "tabby"}};
+        stringmap c     = {{"cat", "siamese"}, {"dog", "lab"}, {"bear", "polar"}};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_multimap
+        using stringmap = std::unordered_multimap<std::string, std::string>;
+        stringmap a     = {{"cat", "tabby"}, {"dog", "poodle"}, {"cat", "siamese"}, {"dog", "poodle"}};
+        stringmap b     = {{"dog", "poodle"}, {"cat", "siamese"}, {"cat", "tabby"}, {"dog", "poodle"}};
+        stringmap c     = {{"cat", "siamese"}, {"dog", "lab"}, {"bear", "polar"}};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_set
+        std::unordered_set<std::string> a = {"cat", "dog", "bear"};
+        std::unordered_set<std::string> b = {"bear", "cat", "dog"};
+        std::unordered_set<std::string> c = {"mouse", "cat", "bear", "dog"};
+        unordered_containers_test(a, b, c);
+    }
+    { // unordered_multiset
+        std::unordered_set<std::string> a = {"cat", "dog", "cat"};
+        std::unordered_set<std::string> b = {"cat", "cat", "dog"};
+        std::unordered_set<std::string> c = {"mouse", "cat", "bear", "dog"};
+        unordered_containers_test(a, b, c);
+    }

```

Comment:
The comment says `unordered_multiset`, but it's still testing `unordered_set`. (This compiles and runs because the non-multi "unique" set is perfectly happy to be constructed from `{"cat", "dog", "cat"}`, it'll just deduplicate the cats :scream_cat: )